### PR TITLE
Docs: remove build warnings and add FP8 caching note

### PR DESCRIPTION
**Pull request by ksivaman**
Signed-off-by: Kirthi Shankar Sivamani <ksivamani@nvidia.com>

### CODE REVIEW

File: docs/examples/advanced_optimizations.ipynb [hunk 1]
```diff
@@ -264,7 +265,15 @@
...
+    "Since weights are typically trained in FP32, a type conversion is required before we can perform compute in FP8. By default, the [fp8_autocast](../api/pytorch.rst#transformer_engine.pytorch.fp8_autocast) context manager will handle this internally by casting non-FP8 tensors to FP8 as they are encountered. However, we can improve upon this in some cases. In particular, if our training iteration is split into multiple gradient accumulation steps, each micro-batch will encounter the same weight tensors. Thus, we only need to cast the weights to FP8 in the first gradient accumulation step and we can cache the resulting FP8 weights for the remaining gradient accumulation steps.\n",
+    "\n",
+    "<div class=\"alert alert-warning\">\n",
+    "\n",
+    "<b>Warning!</b> \n",
+    "\n",
+    "The precise numerical outputs with and without the FP8 weight caching optimization may not be bitwise identical. This is because while the weights remain frozen across a gradient accumulation cycle, the scaling factors and amaxes for the FP8 weights can change as they are updated at the end of every iteration.\n",
```
Comment (timmoon10) [inline docs/examples/advanced_optimizations.ipynb:274]: If we want to be pedantic, it's the scaling factors and amax histories that are affected? The amaxes themselves should be consistent.

**Review by timmoon10 — APPROVED** [review]

File: docs/examples/advanced_optimizations.ipynb (hunk 1 above)
Comment (ksivaman) [reply docs/examples/advanced_optimizations.ipynb:274]: The amax history is just the storage of the current/previous amaxes. Although you're right, a good point would be to include that the history is not frozen with the weights.

**Pull request merged**